R/xgx_stat_ci.R

```diff
@@ -101,85 +101,203 @@
 #' @importFrom ggplot2 aes
 #' @importFrom ggplot2 position_dodge
 #' @export
-xgx_stat_ci <- function(mapping = NULL, data = NULL, conf_level = 0.95,
+xgx_stat_ci <- function(mapping = NULL,
+                        data = NULL,
+                        conf_level = 0.95,
                         distribution = "normal",
+                        bins = NULL,
+                        breaks = NULL,
                         geom = list("point", "line", "errorbar"),
                         position = "identity",
                         fun.args = list(),
+                        fun.data = NULL,
                         na.rm = FALSE,
                         show.legend = NA,
                         inherit.aes = TRUE,
                         ...) {
-  if (!(conf_level > 0.5 && conf_level < 1)) {
-    stop("conf_level should be greater than 0.5 and less than 1")
+
+  lays <- list()
+
+  # Confidence intervals via `xgx_conf_int` is the default function
+  if (is.null(fun.data)) {
+    fun.data <- function(y) xgx_conf_int(y = y,conf_level = conf_level,
+                                         distribution = distribution)
```

Comment:
Does this work for ordinal / multinomial? Or does it give the wrong y-axis again? I haven't checked

Reply:
Or maybe you need to have a StatSummaryOrdinal to deal with the ordinal case (grouping by color, fill, panel, etc. and summarizing the count as length of x within the groups)? Similar to the StatSummaryBinOrdinal but without the bins?